src/listeners/decorators/on.decorator.ts

```diff
@@ -0,0 +1,11 @@
+import { Listener } from 'necord';
+import { NecordLavalinkEvents } from '../listener.interface';
+
+/**
+ * Decorator that marks a method as a listener for Lavalink client.
+ * @param event The event name.
+ * @returns The decorated method.
+ * @url https://necord.org/recipes/lavalink/listeners
+ */
+export const OnLavalink = <K extends keyof E, E = NecordLavalinkEvents>(event: K) =>
+	Listener({ type: 'on', event });
```

Comment:
Can we here `On` ? It also have generic types

Reply:
Sorry I didn't understand the question

Reply:
I mean can we use `@On` decorator instead of `@Listener` from @necord?

Reply:
From the discussion below, we can replace decorators from necord by own

Reply:
Could you show me an example?